src/users/users.controller.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Param,
+  Query,
+  Delete,
+  Put,
+} from '@nestjs/common';
+import { ApiBody, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { UsersService } from './users.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { SearchUserDto } from './dto/search-user.dto';
+import { User } from './users.model';
+
+@ApiTags('Users')
+@Controller('v1/users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @ApiOperation({ summary: 'User creation' })
+  @ApiResponse({ status: 200, type: User })
+  @ApiBody({ type: CreateUserDto })
+  @Post()
+  async create(@Body() createUserDto: CreateUserDto) {
+    return await this.usersService.createUser(createUserDto);
+  }
+
+  @ApiOperation({ summary: 'Get all users' })
+  @ApiResponse({ status: 200, type: [User] })
+  @Get()
+  async findAll(@Query() query: SearchUserDto) {
+    return await this.usersService.findAll(query);
+  }
+
+  @ApiOperation({ summary: 'Get one user by id' })
+  @ApiResponse({ status: 200, type: User })
+  @Get(':id')
+  async findOne(@Param('id') id: string) {
+    return await this.usersService.findOne(id);
+  }
+
+  @ApiOperation({ summary: 'Update user' })
+  @ApiResponse({ status: 200, type: User })
+  @Put(':id')
+  async update(@Param('id') id: string, @Body() updateUserDto: UpdateUserDto) {
+    return await this.usersService.update(id, updateUserDto);
+  }
+
+  @ApiOperation({ summary: 'Remove user' })
+  @ApiResponse({ status: 200, type: User })
```

Comment:
For delete usually use 204 NO CONTENT. IMHO soft delete hasn't to send user body in the same way